### PR TITLE
daemon: Write persistence when Y=0 for DPI X only devices

### LINKED ISSUE #1744: DeathAdder 3.5G doesn't persist DPI across reboots
> Originally reported in https://github.com/polychromatic/polychromatic/issues/404:

The DeathAdder 3.5G (a single DPI X device) keeps resetting its DPI value to 3500 across reboots. 

`~/.config/openrazer/persistence.conf` does not contain any `dpi` entries, so it might be silently failing to save the value (being the odd one with no Y value). Possibly that could be reason it resets because the persistence value defaults to `0`.

Manually writing the desired DPI to file and setting it as read only was confirmed to work:
```
[DeathAdder35G]
poll_rate = 1000
dpi_x = 900
dpi_y = 0
```

### PR DESCRIPTION
**Pull request by codefiles**
Fixes #1744

### CODE REVIEW

File: daemon/openrazer_daemon/daemon.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ def write_persistence(self, persistence_file):
             if 'set_dpi_xy' in device.dbus.METHODS or 'set_dpi_xy_byte' in device.dbus.METHODS:
                 dpi_x = int(device.dbus.dpi[0])
                 dpi_y = int(device.dbus.dpi[1])
-                if dpi_x > 0 and dpi_y > 0:
+                if dpi_x > 0 and dpi_y > 0 or 'available_dpi' in device.dbus.METHODS and dpi_y == 0:
```
Comment (z3ntu) [inline daemon/openrazer_daemon/daemon.py]: I think this should also work?
```python
# Devices with 'available_dpi' only have X values, Y will be 0
if dpi_x > 0 and (dpi_y > 0 or 'available_dpi' in device.dbus.METHODS):
```

Comment (codefiles) [reply daemon/openrazer_daemon/daemon.py]: That works if the check for `dpi_y` being equal to zero is not needed. Zero is the only valid `dpi_y` value for a DPI X only device. Without that check any invalid `dpi_y` value (negative values) for DPI X only devices could be saved to the persistence file. I'm not sure how the value could end up invalid but we don't have to find out if we keep the check. The additional check has no impact on performance.

Comment (z3ntu) [reply daemon/openrazer_daemon/daemon.py]: Without checking Python docs how an `and` - `or` chain gets evaluated I'd like to see some brackets in your solution to make it more obvious which `and` & `or` groups belong together. And add a comment above also to explain what this line should do, and the specifics about devices with `available_dpi`

**Review by z3ntu — APPROVED** [review]
LGTM, thanks!

I won't say that it's now super easy to understand but good enough, and probably best for performance (not that it matters much here) :)

**Pull request merged**